src/ORM/DataObject.php

```diff
@@ -4186,6 +4187,31 @@ public function mergeRelatedObjects($list, $items)
         return $added;
     }
 
+    /**
+     * Generate a unique key for data object
+     * the unique key uses the @see DataObject::getCacheKeyComponent() extension point so unique key modifiers
+     * such as versioned or fluent are covered
+     * i.e. same data object in different stages or different locales will produce different unique key
+     *
+     * recommended use:
+     * - when you need unique key for caching purposes
+     * - when you need unique id on the front end (for example JavaScript needs to target specific element)
+     *
+     * @return string
+     */
+    public function getUniqueKey(): string
+    {
+        /** @var UniqueKeyInterface $service */
+        $service = UniqueKeyService::singleton();
```

Comment:
Using a service for providing the unique key *will* have a performance impact. Is it negligible or significant? I imagine that this could easily be called 1000s of times when generating a page.

If it's not negligible, are we confident that the benefits of making the generator pluggable are justified? Can we envision use-cases for a custom generator?

Reply:
It's using the service as singleton so I don't think this would cause performance issues. Btw this is not a new feature but a code reuse from a massive project where poor performance snowballs and becomes a problem really fast. I'm pretty confident that this will be alright ;-)